### PR TITLE
Update Twurple->7.0.0.

### LINKED ISSUE #126: Update Twurple to 7.0.0
This will be a little work, but they provide a [migration guide](https://twurple.js.org/docs/migration/). I'll be picking this up ASAP.

### PR DESCRIPTION
**Pull request by demize**
This is effectively an alertnate method to address https://github.com/fluid-queue/fluid-queue/pull/115, which keeps the
same logic we were using before that PR. As such, that PR was reverted
prior to implementing this upgrade.

Additionally, this updates some dev dependencies that themselves depend
on `semver` to fix a vulnerability there. It doesn't impact us, but it
needs to be updated anyway.

### Checklist

- [X] Have you followed the guidelines in our Contributing document?
- [X] Have you run `eslint` on the code and resolved any errors?
- [X] Have you run `npm test` and all tests are passing?
- [ ] Have you added new tests for any new functions created?
- [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you updated CHANGELOG.md to add your changed under the `[Unreleased]` heading?

### Benefits

More ergonomic usage of certain APIs, a fully up-to-date Twitch library, and the fix to the channel points scope issue re #114.

Closes #126.

### Potential drawbacks

While the changes overall are minimal, there could be regressions caused by the upgrade that we haven't caught yet. Merging this early is important so we can detect regressions in actual use prior to a full release of v2.0.0.

### CODE REVIEW

**Comment by demize** [comment]
@liquidnya if you could test that #115 is still fixed here before approving, I'd appreciate it, you're probably better set up to test that than I am.

**Comment by liquidnya** [comment]
> @liquidnya if you could test that https://github.com/fluid-queue/fluid-queue/pull/115 is still fixed here before approving, I'd appreciate it, you're probably better set up to test that than I am.

@demize I tested this branch (`ace20c2269d85ab5f6faba172ef68391c2cb809c`) with the following setups:

- Broadcaster token with scopes: channel:manage:redemptions channel:read:redemptions channel:read:subscriptions moderation:read moderator:read:chatters
Bot token with scopes: chat:edit chat:read
- Bot (that is not the broadcaster) token with the following scopes:
chat:read chat:edit moderator:read:chatters
- Bot = Broadcaster token with the scopes:
channel:manage:redemptions channel:read:redemptions channel:read:subscriptions chat:edit chat:read moderation:read moderator:read:chatters

So it works with the same tokens that I used to test #115.
I did not test that `channel:read:redemptions` can be omitted.

**Comment by demize** [comment]
> I did not test that `channel:read:redemptions` can be omitted.

Not necessary for this PR, since the channel point logic is still sitting patiently in its own PR. This will fix it there once I rebase that branch.

File: src/twitch-api.ts [hunk 1]
```diff
@@ -73,6 +73,13 @@ class TwitchApi {
     return this.#broadcasterUser;
   }
 
+  private get moderator(): UserIdResolvable {
+    if (this.#botUserId == null) {
+      throw new Error("Tried to load chatters before client set up");
+    }
+    return this.#botUserId;
+  }
+
```
Comment (liquidnya) [inline src/twitch-api.ts:82]: We can remove this method as it is not used anywhere anymore and also the name is misleading.

**Review by liquidnya — CHANGES_REQUESTED** [review]
LGTM

File: src/twitch-api.ts (hunk 1 above)
Comment (demize) [reply src/twitch-api.ts]: Done.

**Review by liquidnya — APPROVED** [review]

**Pull request merged**